recipes/libpq/all/conanfile.py

```diff
@@ -156,12 +179,44 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
+    def _construct_library_name(self, name):
+        if self.settings.compiler == "Visual Studio":
+            return "lib{}".format(name)
+        return  name
+
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "PostgreSQL"
         self.cpp_info.names["cmake_find_package_multi"] = "PostgreSQL"
         self.env_info.PostgreSQL_ROOT = self.package_folder
-        self.cpp_info.libs = tools.collect_libs(self)
+        
+        self.cpp_info.components["pq"].libs = [self._construct_library_name("pq")]
```

Comment:
Components!!!

Reply:
Yes thanks to components we can handle the mess of those libraries that depend of version, compiler dans link... 🤔